code/eyetracking/eyetrackingrun.py

```diff
@@ -0,0 +1,781 @@
+        """
+
+        row_start = self.messages[
+            self.messages["trialid "].str.contains("RECORD", case=False, regex=True)
+        ].head(1)
+        start_text = row_start["trialid "].iloc[0]
+        match_start = re.search(r"RECORD (\w+) (\d+) (\d+) (\d+) (\w+)", start_text)
+
+        if match_start:
+            eye_tracking_method, sampling_frequency, _, _,r_eye = match_start.groups()
+            sampling_frequency = int(sampling_frequency)
+            if r_eye=='R':
+                recorded_eye="right"
+            elif r_eye=='L':
+                recorded_eye = "right"
+            elif r_eye=='RL':
+                recorded_eye = "both"
+            else:
+                recorded_eye = "unknown"
+
+            print("recorded eye:",recorded_eye)
+            print("Eye Tracking Method:", eye_tracking_method)
+            print("Sampling Frequency:", sampling_frequency)
+        else:
+            eye_tracking_method = "unknown"
+            sampling_frequency = None
+            print("Eye Tracking Method: unknown")
+            print("Sampling Frequency: Not available")
+
+        row_thresholds = self.messages[
+            self.messages["trialid "].str.contains("THRESHOLDS", case=False, regex=True)
+        ].head(1)
+        thresholds_text = row_thresholds["trialid "].iloc[0]
+        print(thresholds_text, "threshold")
+        match_thresholds = re.search(r"THRESHOLDS (\w+) (\d+) (\d+)", thresholds_text)
+
+        if match_thresholds:
+            _, pupil_threshold, CR_threshold = match_thresholds.groups()
+            pupil_threshold = int(pupil_threshold)
+            CR_threshold = int(CR_threshold)
+            print("Pupil Threshold:", pupil_threshold)
+            print("CR Threshold:", CR_threshold)
+        else:
+            pupil_threshold = None
+            CR_threshold = None
+            print("Pupil Threshold: Not available")
+            print("CR Threshold: Not available")
+
+        row_fit_param = self.messages[
+            self.messages["trialid "].str.contains("ELCL_PROC", case=False, regex=True)
+        ].head(1)
+        fit_param_text = row_fit_param["trialid "].iloc[0]
+
+        if "ELLIPSE" in fit_param_text:
+            pupil_fit_method = "ellipse"
+        else:
+            pupil_fit_method = "center-of-mass"
+
+        print("Pupil Fitting Method:", pupil_fit_method)
+
+        return (
+            recorded_eye,
+            eye_tracking_method,
+            sampling_frequency,
+            pupil_threshold,
+            CR_threshold,
+            pupil_fit_method,
+        )
+
+    def extract_header(self) -> List[str]:
+        """
+        Extracts header information from the messages DataFrame.
+
+        Returns:
+        list[str]: A list of strings containing the extracted header information.
+        """
+        self.messages["trialid_cleaned"] = self.messages["trialid "].str.replace(
+            r"[\n\x00\t]", ""
+        )
```

Comment:
This would better be done with `string` as I mentioned to you (and gave you the example of the physio notebook where this cleanup is also done).

```suggestion
        self.messages["trialid_cleaned"] = self.messages["trialid "].str.replace(
            r"[\n\x00\t]", ""
        )
```

Reply:
This remains outstanding.